src/FsCheck/Testable.fs

```diff
@@ -32,35 +24,105 @@ type Result =
         | (Outcome.True,_) -> r
         | (Outcome.Rejected,Outcome.Rejected) -> l //or r, whatever
     static member (|||) (l,r) =
-        match (l.Outcome, r.Outcome) with
+        match (l,r) with
         | (Outcome.Exception _,_) -> r
         | (_,Outcome.Exception _) -> l
         | (Outcome.Timeout _,_) -> l
         | (_,Outcome.Timeout _) -> r
         | (_,Outcome.False) -> l
         | (Outcome.False,_) -> r
         | (Outcome.True,_) -> l
-        | (_,Outcome.True) -> r
+        | (_,Outcome.True) -> r        
         | (Outcome.Rejected,Outcome.Rejected) -> l //or r, whatever
 
+type OutcomeContainer = 
+    | Value of Outcome
+    | Future of Threading.Tasks.Task<Outcome>
+
+///The result of one execution of a property.
+[<NoComparison>]
+type Result = 
+    {   Outcome     : OutcomeContainer
+        Stamp       : list<string>
+        Labels      : Set<string>
+        Arguments   : list<obj> } 
+        
+    static member private combine f lr (lo :'a) rr (ro :'a) = 
+        let x = f lo ro
+        if x = lo then lr else rr
+
+    static member private combineArray f xs = 
+        match xs with
+        | [|lo;ro|] -> f lo ro
+        | _ -> failwith "no way"
+
+    ///Returns a new result that is Succeeded if and only if both this
+    ///and the given Result are Succeeded.
+
+    static member (&&&) (l,r) = 
+        //printfn "And of l %A and r %A" l.Outcome r.Outcome
+        match (l.Outcome,r.Outcome) with
+        | (OutcomeContainer.Future tl,OutcomeContainer.Future tr) -> 
+            { 
+            Outcome = OutcomeContainer.Future <| Threading.Tasks.Task.WhenAll([|tl; tr|]).ContinueWith(fun (xs :Threading.Tasks.Task<Outcome[]>) -> Result.combineArray (&&&) xs.Result)
+            Stamp = [] // that superlame I'm afraid I had to place task at `Value of Result / Future of Task<Result>` level 
```

Comment:
> Perhaps the OutcomeContainer should be a ResultContainer so the whole thing can be wrapped in a future; or we can use blocking on Result with the only downside (?) being that users of &&& and ||| will be potentially slowed down.

That exactly the options which I'm considering for now. I'm afraid wrapping result in a future may end up with bigger mess and code duplication due to "async all the way down", and maybe we would found ourselves in front of another call requiring blocking access somwhere up the call chain. ☹️  
To block at `&&&` and `|||` seems seductive for me but somewhat unfair regarding users, so I'll try to hack quick solution with first approach and left blocking as fallback option.